### PR TITLE
Line of sight

### PR DESCRIPTION
**Pull request by ravifrancesco**
The class LineOFSight.java provides a method to determine which POIPoints are visible out of an input list of POIPoints

### CODE REVIEW

File: app/src/main/java/com/github/ravifrancesco/softwaredevproject/LineOfSight.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+ *
+ * The ELEVATION_DIFFERENCE_THRESHOLD represents the maximum acceptable difference in meters
+ * between the line that connects the user to the POIPoint and the actual elevation of the
+ * terrain in a given point.
+ */
+public class LineOfSight {
+
+    static final int ELEVATION_DIFFERENCE_THRESHOLD = 50; // in meters
+
+    private UserPoint userPoint;
```
Comment (magangloff) [inline app/src/main/java/com/github/ravifrancesco/softwaredevproject/LineOfSight.java:21]: userPoint can be made final
```suggestion
    private final UserPoint userPoint;
```

File: app/src/main/java/com/github/ravifrancesco/softwaredevproject/LineOfSight.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+    private ElevationMap elevationMap;
```
Comment (magangloff) [inline app/src/main/java/com/github/ravifrancesco/softwaredevproject/LineOfSight.java:23]: elevationMap can be made final
```suggestion
    private final ElevationMap elevationMap;
```

File: app/src/main/java/com/github/ravifrancesco/softwaredevproject/LineOfSight.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+            line.add(new Pair<>(x, y));
+
+            // Add slope to increment angle formed
+            slope_error_new += m_new;
+
+            // Slope error reached limit, time to
+            // increment y and update slope error.
+            if (slope_error_new >= 0)
+            {
```
Comment (magangloff) [inline app/src/main/java/com/github/ravifrancesco/softwaredevproject/LineOfSight.java:122]: Brackets should be the same everywhere
```suggestion
            if (slope_error_new >= 0){
```

**Review by magangloff — APPROVED** [review]
Small comments, else all good

File: app/src/androidTest/java/com/github/ravifrancesco/softwaredevproject/ElevationMapTest.java [hunk 4]
```diff
@@ -76,4 +80,88 @@ public void updateMapTest() {
...
+        UserPoint userPoint = new UserPoint(mContext);
+        userPoint.setLocation(GPSTracker.DEFAULT_LAT, GPSTracker.DEFAULT_LON,GPSTracker.DEFAULT_ALT, GPSTracker.DEFAULT_ACC);
+
+        ElevationMap elevationMap = new ElevationMap(userPoint);
+
+        Assert.assertEquals(0.000833333333, elevationMap.getMapCellSize(), 0.00000001);
+
+
+    }
+
```
Comment (bgabriel998) [inline app/src/androidTest/java/com/github/ravifrancesco/softwaredevproject/ElevationMapTest.java:150]: Add test comment

File: app/src/androidTest/java/com/github/ravifrancesco/softwaredevproject/ElevationMapTest.java [hunk 5]
```diff
@@ -76,4 +80,88 @@ public void updateMapTest() {
...
+        elevationMap.updateElevationMatrix();
+
+        // check the altitude around the Mont Blanc Peak using coordinates
+        Assert.assertEquals(4808, elevationMap.getAltitudeAtLocation(45.8326, 6.8652), 200);
+        // check the altitude around the Mont Blanc Peak using indexes
+        indexes = elevationMap.getIndexesFromCoordinates(45.8326, 6.8652);
+        Assert.assertEquals(4808, elevationMap.getAltitudeAtLocation(indexes.first, indexes.second), 200);
+
+    }
+
```
Comment (bgabriel998) [inline app/src/androidTest/java/com/github/ravifrancesco/softwaredevproject/ElevationMapTest.java:135]: Add test comment

File: app/src/androidTest/java/com/github/ravifrancesco/softwaredevproject/LineOfSightTest.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+import org.osmdroid.util.GeoPoint;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+public class LineOfSightTest {
+
```
Comment (bgabriel998) [inline app/src/androidTest/java/com/github/ravifrancesco/softwaredevproject/LineOfSightTest.java:18]: Add test comment

File: app/src/androidTest/java/com/github/ravifrancesco/softwaredevproject/LineOfSightTest.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        pointsToCheck.add(point1);
+        pointsToCheck.add(point2);
+        pointsToCheck.add(point3);
+        pointsToCheck.add(point4);
+
+        Assert.assertEquals(new HashSet<POIPoint>(lineOfSight.getVisiblePoints(pointsToCheck)), new HashSet<POIPoint>(visiblePoints));
+
+    }
+
```
Comment (bgabriel998) [inline app/src/androidTest/java/com/github/ravifrancesco/softwaredevproject/LineOfSightTest.java:56]: Add test comment

**Review by bgabriel998 — COMMENTED** [review]
Please add the requested comments in test file. 
Otherwise, good job. It's nice that you have used FP statements and parallelization

**Review by bgabriel998 — APPROVED** [review]

**Pull request merged**